### ISSUE TITLE
Crashes during striking fleet sortie sending

### ISSUE THREAD

**Issue by stackhanovets** [perryhuynh/kcauto#97]
So I've just updated the tools:

```
git clean -dn && git clean -df
git pull origin 64-event-2018-winter
cd .. 
wget https://launchpad.net/sikuli/sikulix/1.1.2/+download/sikulixsetup-1.1.2.jar
rm -rf /tmp/Sikuli*
java -jar sikulixsetup-1.1.2.jar
...
1, 1a & 2
...
```
The script has launched succesfully, took control & crashed just on the pre-sortie screen with, um, `combat_start_lbas.png`:

![bt](https://user-images.githubusercontent.com/16455270/37152133-5d3d2996-22e9-11e8-9c7c-2f2e21190893.png)

```
sed 's|[0-9]\+-[0-9]\+-[0-9]\+|date|g' log | sed 's|[0-9]\+:[0-9]\+:[0-9]\+|time|g' | sed 's|[0-9]\+x[0-9]\+|resolution|g' | sed 's|[0-9]\+:Chrome|Chrome|'
```
```
[date time] At side menu
[date time] Going home.
[log] CLICK on L(78,373)@S(0)[0,0 resolution] (547 msec)
[date time] At home
[date time] At home
[date time] Navigating to combat screen.
[log] CLICK on L(186,378)@S(0)[0,0 resolution] (576 msec)
[log] CLICK on L(241,372)@S(0)[0,0 resolution] (536 msec)
[log] CLICK on L(716,561)@S(0)[0,0 resolution] (534 msec)
[date time] Begin resupplying LBAS groups.
[log] CLICK on L(235,505)@S(0)[0,0 resolution] (555 msec)
[date time] Resupplying LBAS group 1.
[date time] LBAS Group 1 fatigue state: Rested
[date time] LBAS group switched to sortie mode.
[date time] Resupplying LBAS group 2.
[log] CLICK on L(691,234)@S(0)[0,0 resolution] (534 msec)
[date time] LBAS Group 2 fatigue state: Rested
[date time] LBAS group switched to sortie mode.
[log] CLICK on L(218,503)@S(0)[0,0 resolution] (559 msec)
[log] CLICK on L(755,385)@S(0)[0,0 resolution] (550 msec)
[log] CLICK on L(658,359)@S(0)[0,0 resolution] (549 msec)
[log] CLICK on L(712,558)@S(0)[0,0 resolution] (551 msec)
[log] CLICK on L(420,229)@S(0)[0,0 resolution] (543 msec)
[error] script [.../kcauto-kai.sikuli ] stopped with error in line 65
[error] KeyError ( 'game' )
[error] --- Traceback --- error source first
line: module ( function ) statement 
361: util (  rejigger_mouse )     rand_x = regions['game'].x + cls.randint_gauss(x1, x2)
382: util (  click_preset_region )     cls.rejigger_mouse(regions, preset)
1073: combat (  check_damages_7th )     Util.click_preset_region(regions, '7th_next')
306: combat (  _run_pre_sortie_fleet_check_logic )     if self.config.combat['fleet_mode'] == 'striking'
229: combat (  _conduct_pre_sortie_checks )     needs_resupply, self.dmg, fleet_fatigue = (
110: combat (  combat_logic_wrapper )     if self._conduct_pre_sortie_checks():
325: main (  run_combat_cycle )     if self.modules['combat'].combat_logic_wrapper():
65: main (  <module> )     kcauto_kai.run_combat_cycle()
[error] --- Traceback --- end --------------

```
CFG:
```
[Combat]
Enabled: True
Engine: legacy
Map: E-5
FleetMode: striking
CombatNodes: 5
NodeSelects: 
Formations: 1:diamond,2:line_ahead,3:line_ahead,4:line_ahead
NightBattles: 1:False,2:False,3:False,4:True
RetreatLimit: heavy
RepairLimit: moderate
RepairTimeLimit: 0030
LBASGroups: 1,2
LBASGroup1Nodes: O,N
LBASGroup2Nodes: T,T
LBASGroup3Nodes: 
MiscOptions: CheckFatigue
```
I see you're having fun with multiple branches - whatever, I do not care. The rollback is:
```
git reset --hard 388fe0a585aa3cdf9048b10bae17c8a815fd6d4a
```
After manual merging with required JSONs and PNGs there were no crashes during the pre-sortie checkout step.

**Comment by waicool20** [comment]
Pull winter branch and checkout `combat.py` from staging or  insert this line after line 57 ```            'game': self.kc_region,  ```

**Comment by perryhuynh** [comment]
You did all this when you could have just done `git checkout staging combat.py`...? yikes

**Comment by stackhanovets** [comment]
Did not make a sense as well the replacement with https://raw.githubusercontent.com/mrmin123/kcauto-kai/staging/kcauto-kai.sikuli/combat.sikuli/combat.py

**Comment by mrmin123** [comment]
As others have noted, the fix for this is already in `staging`, which will be merged into `master` and the event branch after #63 and #87 are completed.

**Comment by stackhanovets** [comment]
Broken `master`? Nice.

**Comment by mrmin123** [comment]
Yes, broken `master`. I'm not sure if you're new to OSS but this happens all the time, not just in this project. If you're expecting more from 1 developer who does this in his spare time, you can either 1. use another project, or 2. kindly fork off.

**Comment by stackhanovets** [comment]
Nice try man, but I'm here to test your stuff. You may be afraid of more stupid issues - I do not care.

**Comment by waicool20** [comment]
`master` branch does not imply its 100% working lul, no software is 100% bug free. If you wanted the latest bug fixes and features then Git good and learn how to checkout files properly. `staging` branch might have fixes for `master` branch bugs, but it itself will have bugs. 

If you like testing, then good but don't complain when you find something broken. That's the whole point of testing :man_facepalming:

**Comment by stackhanovets** [comment]
Yeah. Creating as much branches as possible is the point. Whatever.
